src/Microsoft.AspNetCore.Mvc.Razor.Extensions/InstrumentationPass.cs

```diff
@@ -93,7 +93,7 @@ public override void VisitHtml(HtmlContentIRNode node)
 
             public override void VisitCSharpExpression(CSharpExpressionIRNode node)
             {
-                if (node.Source != null && !(node.Parent is CSharpAttributeValueIRNode))
+                if (node.Source != null && !(node.Parent is CSharpExpressionAttributeValueIRNode))
                 {
                     Items.Add(new InstrumentationItem(node, isLiteral: false, source: node.Source.Value));
                 }
```

Comment:
It can't you are right. I did this change before I changed some more things in lowering phase. Will revert.